src/xpk/core/blueprint/blueprint_generator.py

```diff
@@ -354,6 +359,268 @@ def _get_a3_mega_blueprint_dependencies(
+                "allow": [
+                    {"protocol": "tcp", "ports": ["0-65535"]},
+                    {"protocol": "udp", "ports": ["0-65535"]},
+                    {"protocol": "icmp"},
+                ],
+            }],
+        },
+    )
+    rma_net_id = f"{cluster_name}-a3u-rdma-net"
+    rma_net = DeploymentModule(
+        id=rma_net_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//community/modules/network/rdma-vpc?ref=98c49fe",
+        settings={
+            "network_name": f"{cluster_name}-a3u-rdma-net",
+            "mtu": mtu_size,
+            "network_profile": f"https://www.googleapis.com/compute/beta/projects/{project_id}/global/networkProfiles/{zone}-vpc-roce",
+            "network_routing_mode": "REGIONAL",
+            "subnetworks_template": {
+                "name_prefix": f"{cluster_name}-a3u-rdma-sub",
+                "count": 8,
+                "ip_range": "192.168.128.0/18",
+                "region": region,
+            },
+        },
+    )
+    cluster_id = f"{cluster_name}-a3-ultragpu-cluster"
+    a3_ultra_cluster = DeploymentModule(
+        id=cluster_id,
+        source="github.com/GoogleCloudPlatform/cluster-toolkit.git//modules/scheduler/gke-cluster?ref=e0c690b",
+        use=[net_0_id],
+        settings={
+            "release_channel": "RAPID",
+            "prefix_with_deployment_name": False,
+            "name_suffix": cluster_name,
+            "system_node_pool_machine_type": system_node_pool_machine_type,
+            "enable_dcgm_monitoring": True,
+            "enable_gcsfuse_csi": True,
+            "enable_private_endpoint": False,
+            "master_authorized_networks": [{
+                "cidr_block": auth_cidr,
+                "display_name": "kubectl-access-network",
+            }],
+            "system_node_pool_node_count": {
+                "total_min_nodes": system_node_pool_min_node_count,
+                "total_max_nodes": 1000,
+            },
+            "additional_networks": (
+                f"$(concat([{{network={cluster_name}-a3u-net-1.network_name,"
+                f" subnetwork={cluster_name}-a3u-net-1.subnetwork_name,"
+                f' subnetwork_project="{project_id}", nic_type="GVNIC",'
+                " queue_count=null, network_ip=null, stack_type=null,"
+                " access_config=[{nat_ip=null, public_ptr_domain_name=null,"
+                " network_tier=null}], ipv6_access_config=[],"
+                " alias_ip_range=[]}],"
+                f" {cluster_name}-a3u-rdma-net.subnetwork_interfaces_gke))"
+            ),
+        },
+        outputs=["instructions"],
+    )
+    system, _ = get_system_characteristics_by_device_type(a3ultra_device_type)
+    if system is None:
+      xpk_print(
+          "Error: Could not retrieve system characteristics for"
+          f" {a3ultra_device_type} device_type."
+      )
+      xpk_exit(1)
+    gpu_pool_id = f"{cluster_name}-a3u-pool"
```

Comment:
nit: do we use this variable anywhere else?

Reply:
we dont